### PR TITLE
tyr: Add shape on user

### PR DESCRIPTION
**Pull request by patochectp**

### CODE REVIEW

File: source/tyr/migrations/versions/44dab62dfff7_add_shape_on_user.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"""empty message
```
Comment (kinnou02) [inline source/tyr/migrations/versions/44dab62dfff7_add_shape_on_user.py]: You should explain what this migration do, this text will be shown at the execution of the migration

File: source/tyr/tyr/resources.py [hunk 2]
```diff
@@ -47,10 +48,19 @@
...
 
 __ALL__ = ['Api', 'Instance', 'User', 'Key']
 
 
+class JsonString(fields.Raw):
+    def __init__(self, **kwargs):
+        super(JsonString, self).__init__(**kwargs)
+
+    def format(self, value):
+        return ast.literal_eval(value)
```
Comment (kinnou02) [inline source/tyr/tyr/resources.py]: What do you want to do here?
a simple `json.loads()` isn't enough?

File: source/tyr/tyr/resources.py [hunk 3]
```diff
@@ -514,11 +527,18 @@ def post(self):
         if not billing_plan:
             return ({'error': 'billing plan doesn\'t exist'}, 400)
 
+        if args['shape']:
+            features =  ast.literal_eval(args['shape'])
```
Comment (kinnou02) [inline source/tyr/tyr/resources.py]: Same thing here, a `json.loads` should be enough.
Json look like python dict, but evaluating it as a python object is wrong

**Review by kinnou02 — CHANGES_REQUESTED** [review]
Did you run the tyr test manually? they aren't run by ctest or jenkins currently :(

File: source/tyr/tyr/resources.py [hunk 4]
```diff
@@ -514,11 +527,18 @@ def post(self):
...
+            validation = is_valid(FeatureCollection(features))
```
Comment (datanel) [inline source/tyr/tyr/resources.py]: is this always FeatureCollection or Feature is valid too?

File: source/tyr/tyr/validations.py [hunk 5]
```diff
@@ -1,8 +1,18 @@
...
 
 def datetime_format(value):
     """Parse a valid looking date in the format YYYYmmddTHHmmss"""
 
     return datetime.strptime(value, "%Y%m%dT%H%M%SZ")
 
+def json_format(value):
+
+    if value:
+        features= value.get('features')
```
Comment (datanel) [inline source/tyr/tyr/validations.py:19]: what happen if I post a Feature (not a FeatureCollection) ?

``` json
{
  "type": "Feature",
  "geometry": {
    "type": "Point",
    "coordinates": [125.6, 10.1]
  },
  "properties": {
    "name": "Dinagat Islands"
  }
}
```

This geojosn has no attribute "features"
[http://geojson.org/](http://geojson.org/)

**Comment by datanel** [comment]
When I test with postman or curl
I have the following errors

```
Traceback (most recent call last):
  File "/home/dquintanel/.virtualenv/tyr/lib/python2.7/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/dquintanel/.virtualenv/tyr/lib/python2.7/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/home/dquintanel/.virtualenv/tyr/lib/python2.7/site-packages/flask_restful/__init__.py", line 271, in error_router
    return original_handler(e)
  File "/home/dquintanel/.virtualenv/tyr/lib/python2.7/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/dquintanel/.virtualenv/tyr/lib/python2.7/site-packages/flask_restful/__init__.py", line 268, in error_router
    return self.handle_error(e)
  File "/home/dquintanel/.virtualenv/tyr/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/dquintanel/.virtualenv/tyr/lib/python2.7/site-packages/flask/app.py", line 1478, in full_dispatch_request
    response = self.make_response(rv)
  File "/home/dquintanel/.virtualenv/tyr/lib/python2.7/site-packages/flask/app.py", line 1566, in make_response
    raise ValueError('View function did not return a response')
ValueError: View function did not return a response
```

**Pull request merged**